[Menu] 메뉴 생성 API 개발 및 테스트 코드

anyubinn opened the pull request:
## Description
- 메뉴 생성 API 개발
  - 메뉴 생성 시 해당 매장의 주인만 작성할 수 있도록 로직 구현
  - 다음과 같은 경우 예외 처리
    - 존재하지 않는 매장
    - 로그인한 회원이 매장 사장이 아닌 경우
    - 매장 사장이지만 해당 매장의 사장이 아닌 경우
  
- 메뉴 생성에 대한 테스트 코드 작성
## Changes
- `MenuRequestDto`, `MenuResponseDto` 추가
- `MenuController`, `MenuService`에 메뉴 생성 관련 로직 작성
- `messages.properties`에 메뉴와 관련된 메시지 추가
- `ErrorCode`에 메뉴와 관련된 에러코드 추가
- `StoreRepositoy` 수정
  - `JpaRepositoy` 추가
## Screenshots
- 현재 매장 생성 관련 API가 아직 없어 테스트 코드로 기능 확인
<img width="280" alt="스크린샷 2025-04-24 오후 2 33 27" src="https://github.com/user-attachments/assets/a97d3909-61c3-49ab-8639-3c91de9d45ac" />

anyubinn committed:
Feature[Menu]: 메뉴 생성 기능 추가
1. 해당 유저를 찾을 수 없다면 예외 발생
2. 해당 유저가 매장 주인이 아니라면
3. 해당 가게를 찾을 수 없다면 예외 발생
4. 해당 유저가 해당 매장 주인이 아니라면 예외 발생

anyubinn committed:
Fix[Menu]: createMenu에 리턴값 추가

anyubinn committed:
Test[Menu]: 메뉴 생성에 대한 테스트코드 추가
1. 메뉴 생성에 성공하는 경우
2. 해당 매장 주인이 아닌 경우
3. 회원이 매장 주인이 아닌 경우

nagul2 commented on src/main/java/com/delivery/igo/igo_delivery/common/exception/ErrorCode.java:
```diff
@@ -47,7 +47,11 @@ public enum ErrorCode {
-    CART_ITEM_NOT_FOUND(HttpStatus.NOT_FOUND, "장바구니를 찾을 수 없습니다.");
+    CART_ITEM_NOT_FOUND(HttpStatus.NOT_FOUND, "장바구니를 찾을 수 없습니다."),
+
+    // Menu
+    STORE_NOT_FOUND(HttpStatus.NOT_FOUND, "가게를 찾을 수 없습니다."),
```
사소하지만 여기 주석 Store 추가해서 해주심 좋을 것 같아요

rlqor9688 reviewed (approved):
고생하셨습니다!!

nagul2 commented on src/test/java/com/delivery/igo/igo_delivery/api/menu/service/MenuServiceImplTest.java:
```diff
@@ -0,0 +1,133 @@
+        given(storeRepository.findById(storeId)).willReturn(Optional.of(store));
+        given(menuRepository.save(any())).willReturn(menu);
+
+        MenuResponseDto responseDto = menuService.createMenu(authUser, storeId, requestDto);
+
+        assertNotNull(responseDto);
```
Service 단위 테스트할 때 목으로 설정한 메서드들이 정상적으로 호출 되었는지 확인하는 verify() 함께 작성하는 것이 일반적이라고하니 참고해주시면 좋을 것 같습니다,
아래 테스트코드도 동일합니다.

anyubinn committed:
Refactor[Menu]: PR 피드백 반영
1. ErrorCode 주석 Menu -> Store로 수정
2. MenuServiceImplTest에 verify 추가하여 테스트 코드의 신뢰성 향상
